code/modules/mob/living/simple_animal/abnormality/waw/snow_whites_apple.dm

```diff
@@ -0,0 +1,292 @@
+		TryTeleport()
+
+/mob/living/simple_animal/hostile/abnormality/snow_whites_apple/AttackingTarget(atom/attacked_target)
+	target = attacked_target
+	return OpenFire()
+
+/mob/living/simple_animal/hostile/abnormality/snow_whites_apple/OpenFire()
+	if(client)
+		togglemovement = TRUE
+		switch(chosen_attack)
+			if(1)
+				if(vine_cooldown <= world.time)
+					vinespike()
+		return
+
+	if(vine_cooldown <= world.time)
+		vinespike()
+
+/mob/living/simple_animal/hostile/abnormality/snow_whites_apple/proc/vinespike()
+	for(var/obj/structure/alien/weeds/F0442/W in view(18, src))
+		W.vinespike()
+	vine_cooldown = world.time + vine_cooldown_time
+
+
+//stolen alien weed code
+/obj/structure/alien/weeds/F0442
+	gender = PLURAL
+	name = "bitter flora"
+	desc = "Branches that grow from wilting stems."
+	icon = 'icons/effects/spacevines.dmi'
+	icon_state = "Med1"
+	anchored = TRUE
+	density = FALSE
+	layer = TURF_LAYER
+	plane = FLOOR_PLANE
+	canSmoothWith = null
+	smoothing_flags = NONE
+	smoothing_groups = null
+	base_icon_state = "Med1"
+	max_integrity = 15
+	last_expand = 0 //last world.time this weed expanded
+	growth_cooldown_low = 0
+	var/list/static/ignore_typecache
+	var/list/static/atom_remove_condition
+	color = "#808000"
+
+
+/obj/structure/alien/weeds/F0442/Initialize()
+	. = ..()
+	pixel_x = 0
+	pixel_y = 0
+
+	if(!atom_remove_condition)
+		atom_remove_condition = typecacheof(list(
+			/obj/projectile/ego_bullet/ego_match,
+			/mob/living/simple_animal/hostile/abnormality/helper,))
+
+	if(!blacklisted_turfs)
+		blacklisted_turfs = typecacheof(list(
+			/turf/open/space,
+			/turf/open/chasm,
+			/turf/open/lava,
+			/turf/open/openspace))
+
+	if(!ignore_typecache)
+		ignore_typecache = typecacheof(list(
+			/obj/effect,
+			/mob/dead,
+			/mob/living/simple_animal/hostile/abnormality/snow_whites_apple))
+
+	last_expand = world.time + growth_cooldown_low
+
+/obj/structure/alien/weeds/F0442/set_base_icon()
+	return
+
+/obj/structure/alien/weeds/F0442/expand()
+	var/turf/U = get_turf(src)
+	if(is_type_in_typecache(U, blacklisted_turfs))
+		qdel(src)
+		return FALSE
+
+	for(var/turf/T in U.GetAtmosAdjacentTurfs())
+		if(locate(/obj/structure/alien/weeds/F0442) in T)
+			continue
+
+		if(is_type_in_typecache(T, blacklisted_turfs))
+			continue
+
+		new /obj/structure/alien/weeds/F0442(T)
+	return TRUE
+
+/obj/structure/alien/weeds/F0442/Crossed(atom/movable/AM)
+	. = ..()
+	if(is_type_in_typecache(AM, atom_remove_condition))
+		qdel(src)
+	if(is_type_in_typecache(AM, ignore_typecache))		// Don't want the traps triggered by sparks, ghosts or projectiles.
+		return
+	if(isliving(AM))
+		vine_effect(AM)
+
+/obj/structure/alien/weeds/F0442/proc/vine_effect(mob/living/L)
+	if(ishuman(L))
+		var/mob/living/carbon/human/lonely = L
+		var/obj/item/trimming = lonely.get_active_held_item()
+		var/brooch = lonely.ego_gift_list[BROOCH]
+		var/broochrefine = "[brooch]"
+		if(broochrefine == "Green Stem")
```

Comment:
Use a typepath for this.

Reply:
I remember trying to use a typepath for this but it didnt work. Ill try and fix it again. Just to make sure im not crazy.

Reply:
Okay i fixed it to istype(brooch, /datum/ego_gift/stem)